### DISCUSSION TITLE
Zooz ZSE19 Siren/Chime Sound Switch Question

### DISCUSSION THREAD

**Discussion by GrizzlyAK** [zwave-js/zwave-js-ui#1541] [Q&A]
Hi (I posted this same question on the zwave-js Git discussion board, but it may be more applicable here)

Env: Home Assistant ZWave JS Integration for Websocket connection to zwavejs2mqtt Add On for Control Panel only.

I have the subject device, which presents in HA as a binary switch for the Siren (works perfectly) and a NOTHING for the Chime. Both Siren and Sound Switch are implemented in the device and are independent. The SS section in the zwavejs2mqtt Control Panel provides the following four parameters: Default Volume, Default Tone ID, Play Tone (a list of sounds in the device at 5 seconds only), and Volume (Volume only provides two options: Default and Custom - but you can't enter a number - this could be a CP issue.)

Using the CP, I can select a sound in the Play Tone parameter and sound that Chime, at the volume I set in the Default Volume parameter.

I have two issues:

(1) After setting a Default Volume (which it does play at) and a Default Tone ID, whenever I have to refresh the page (or after some time, I don't know which), these values reset to 100% and 1, respectively. Should they not be static and reflect what is stored in the device? It appears the device reverts as well. What could be going on here?
(2) I'm not sure how exactly to use the Sound Switch in HA in an automation, although the following works to play a tone, I've been unsuccessful in using similar code to set the Default Volume or any other parameters. I also thought that SS allowed for setting Tone and Volume during the call. Perhaps this is a HA shortcoming, idk.

     - service: zwave_js.set_value
        data:
          command_class: '121'
          endpoint: '0'
          property: toneId
          value: 34
        target:
          entity_id: sensor.zooz_zse19_siren_node_status

I realize some of this may not be the result of the library, but I'm just trying to get my head around what's going on and how best to make this work. Any advice would be appreciated. Thanks.

**Comment by robertsLando** [comment]
Please post a silly level log, we need to check what's going on at the low level.

cc @AlCalzone

**Reply by GrizzlyAK** [reply]
OK, disclaimer: I'm 2-days brand new to zwave-js/zwavejs2mqtt. I just upgraded my HA installation from OZW 1.4.

I set logging in CP to Silly. Also set to log only Node 48 (but that must only apply to the log file, not the debug window). I did the following and saw the posted results. It appears it isn't really changing the values in the Node (although, the results when playing the sound (tone + volume) seemed to be what I entered). ??
```
After entering a value of 75 in the defaultVolume field and pressing Send:
========
2021-08-05 09:56:08.090 INFO ZWAVE: Calling api writeValue with args: [
{
nodeId: 48,
commandClass: 121,
endpoint: 0,
property: 'defaultVolume'
},
75,
{},
[length]: 3
]
2021-08-05 09:56:08.096 INFO ZWAVE: Writing 75 to 48-121-0-defaultVolume
========

After entering a value of 32 in the defaultToneId field and pressing Send:
========
2021-08-05 09:56:43.484 INFO ZWAVE: Calling api writeValue with args: [
{
nodeId: 48,
commandClass: 121,
endpoint: 0,
property: 'defaultToneId'
},
32,
{},
[length]: 3
]
2021-08-05 09:56:43.487 INFO ZWAVE: Writing 32 to 48-121-0-defaultToneId
2021-08-05 09:56:44.805 INFO ZWAVE: Success zwave api call writeValue undefined
=========

From node_48.json.txt File Exported from CP Advanced:
=========
    {
      "id": "48-121-0-defaultVolume",
      "nodeId": 48,
      "commandClass": 121,
      "commandClassName": "Sound Switch",
      "endpoint": 0,
      "property": "defaultVolume",
      "propertyName": "defaultVolume",
      "type": "number",
      "readable": true,
      "writeable": true,
      "label": "Default volume",
      "stateless": false,
      "min": 0,
      "max": 100,
      "unit": "%",
      "list": false,
      "value": 50,
      "lastUpdate": 1628186120087,
      "newValue": 75,
      "toUpdate": true
    },
    {
      "id": "48-121-0-defaultToneId",
      "nodeId": 48,
      "commandClass": 121,
      "commandClassName": "Sound Switch",
      "endpoint": 0,
      "property": "defaultToneId",
      "propertyName": "defaultToneId",
      "type": "number",
      "readable": true,
      "writeable": true,
      "label": "Default tone ID",
      "stateless": false,
      "min": 0,
      "max": 254,
      "list": false,
      "value": 34,
      "lastUpdate": 1628186120090,
      "newValue": 32,
      "toUpdate": true
    },
=========
```
but the values I entered still appear on the screen in the CP.

cc @AlCalzone

**Reply by GrizzlyAK** [reply]
UPDATE: I just noticed the "newValue" items in the json, which does reflect what was entered into the CP.  And it does seem to stick at times, but seemingly randomly resets the defaultVolume to 100.

**Reply by GrizzlyAK** [reply]
In the CP, at the top of each Node frame, there are two options for Send Options: Transition Duration and Volume (for SS). Neither appear to have any effect when playing a Tone using 'toneId`.

**Reply by AlCalzone** [reply]
> disclaimer: I'm 2-days brand new

Ok you are forgiven 😉
Please post a [driver log](https://zwave-js.github.io/zwavejs2mqtt/#/troubleshooting/bug_report?id=driver-logs) of what you posted above.

**Reply by GrizzlyAK** [reply]
You want the entire zip file?

**Reply by AlCalzone** [reply]
No, just the driver log where I can see what happens when you do the above steps.

**Comment by GrizzlyAK** [comment]
Here you go.  I didn't realize that what I posted above is only the zw2m log. Daniel sorted me on how to see the driver logs in the debug window. Looking at the file for Node 48, I don't see any driver traffic related to the above zw2m traffic (but I may have missed it). Weird.

[zwavejs_2021-08-05.zip](https://github.com/zwave-js/zwavejs2mqtt/files/6947341/zwavejs_2021-08-05.zip)

**Reply by AlCalzone** [reply]
Do you have any filters on - like for node 48? I see a bunch of traffic that's not "explained" in the log - some of which might be Sound Switch related.

**Comment by GrizzlyAK** [comment]
I don’t currently, but it is possible – I’ve played with that some. However, it probably would have been on (or off) for a long time, not just a couple of minutes. Of course, I can always run another set of tests if you’d like.

**Comment by ORCHORDS** [comment]
Two separate things here, and the source code for Sound Switch CC in node-zwave-js confirms how each one works.

(1) Default Volume / Default Tone ID resetting after a page refresh — this is by design and not a bug. In packages/cc/src/cc/SoundSwitchCC.ts the defaultVolume and defaultToneId values are queried every time the value cache is rebuilt (after reconnect, page reload, or cache TTL expiry). The UI shows whatever the device currently reports. The Zooz ZSE19 stores Play Tone volume in the tone play command itself, not in CC 121 — those configuration parameters are just defaults for future tones. The next refresh shows the device's defaults again. To lock a default, set it via the device's own Configuration CC parameters (Param 1–4 in the ZSE19 manual) — those ARE persisted across reboots.

(2) Calling Sound Switch from HA — your service call is close. The proper payload for CC 121 is:

```yaml
service: zwave_js.set_value
data:
  command_class: '121'
  endpoint: '0'
  property: toneId
  property_key: 'default'
  value: 34
target:
  entity_id: sensor.zooz_zse19_siren_node_status
```

The cleanest path on newer zwave-js / HA is zwave_js.bulk_set_partial_config_parameters (ZSE19 config params) for the defaults, and the in-app CP for ad-hoc playback. toneId alone only triggers the default tone with the device's stored default volume; the CC 121 Tones_Play command takes tone number, volume, and duration in one frame — but the HA service surface only exposes toneId. To set a custom volume per-tone you'd have to either script two sequential zwave_js.set_value calls (which still doesn't compose atomically) or update to a zwave-js version that supports a writable compound toneId + volume value.

For (1), Robert asked for a silly-level log — paste the driver log lines around the SoundSwitch CC Configuration Get → Report cycle when you set the defaults and they reset, and you'll see the device IS returning 100/1, which confirms the device-side reset, not a CP-side bug.
